rust/xaynet-server/src/storage/redis.rs

```diff
@@ -0,0 +1,644 @@
+        // https://redis.io/commands/get
+        // > Get the value of key. If the key does not exist the special value nil is returned.
+        //   An error is returned if the value stored at key is not a string, because GET only
+        //   handles string values.
+        // > Return value
+        //   Bulk string reply: the value of key, or nil when key does not exist.
+        self.connection.get("coordinator_state").await
+    }
+
+    /// Stores a [`CoordinatorState`].
+    ///
+    /// If the coordinator state already exists, it is overwritten.
+    pub async fn set_coordinator_state(mut self, state: &CoordinatorState) -> RedisResult<()> {
+        debug!("set coordinator state");
+        // https://redis.io/commands/set
+        // > Set key to hold the string value. If key already holds a value,
+        //   it is overwritten, regardless of its type.
+        // Possible return value in our case:
+        // > Simple string reply: OK if SET was executed correctly.
+        self.connection.set("coordinator_state", state).await
+    }
+
+    /// Retrieves the [`SumDict`].
+    pub async fn get_sum_dict(mut self) -> RedisResult<SumDict> {
+        debug!("get sum dictionary");
+        // https://redis.io/commands/hgetall
+        // > Return value
+        //   Array reply: list of fields and their values stored in the hash, or an empty
+        //   list when key does not exist.
+        let result: Vec<(PublicSigningKeyRead, PublicEncryptKeyRead)> =
+            self.connection.hgetall("sum_dict").await?;
+        let sum_dict = result
+            .into_iter()
+            .map(|(pk, ephm_pk)| (pk.into(), ephm_pk.into()))
+            .collect();
+
+        Ok(sum_dict)
+    }
+
+    /// Stores a new [`SumDict`] entry.
+    ///
+    /// Returns [`AddSumParticipant::Ok`] if field is a new or
+    /// [`AddSumParticipant::AlreadyExist`] if field already exists.
+    pub async fn add_sum_participant(
+        mut self,
+        pk: &SumParticipantPublicKey,
+        ephm_pk: &SumParticipantEphemeralPublicKey,
+    ) -> RedisResult<AddSumParticipant> {
+        debug!("add sum participant with pk {:?}", pk);
+        // https://redis.io/commands/hsetnx
+        // > If field already exists, this operation has no effect.
+        // > Return value
+        //   Integer reply, specifically:
+        //   1 if field is a new field in the hash and value was set.
+        //   0 if field already exists in the hash and no operation was performed.
+        self.connection
+            .hset_nx(
+                "sum_dict",
+                PublicSigningKeyWrite::from(pk),
+                PublicEncryptKeyWrite::from(ephm_pk),
+            )
+            .await
+    }
+
+    /// Removes an entry in the [`SumDict`].
+    ///
+    /// Returns [`DeleteSumParticipant::Ok`] if field was deleted or
+    /// [`DeleteSumParticipant::DoesNotExist`] if field does not exist.
+    pub async fn remove_sum_dict_entry(
```

Comment:
Do we actually need to remove participants?

Reply:
We remove the sum participant in the sum2 phase 
https://github.com/xaynetwork/xaynet/pull/519/files#diff-5b85450e9124bca33ff393eee368f61fR187
My idea was to add this command to the `incr_mask_count` method. However, we will then have the same issue that we have with the `update_seed_dict` method. (the mask count would increase even if the sum particpant is not in the sum dict)